aea/protocols/generator/validate.py

```diff
@@ -685,6 +696,39 @@ def _validate_termination(
                 ),
             )
 
+    # check that there are no repetitive performatives in termination
+    duplicates_dict: Dict[str, int] = {}
+    for performative in termination:
+        duplicates_dict[performative] = duplicates_dict.get(performative, 0) + 1
+    duplicates = [k for k, v in duplicates_dict.items() if v > 1]
```

Comment:
just use `no_duplicates = len(termination) == len(list(set(termination)))`, this is very non-pythonic

Reply:
Yea, I did something similar initially, but this gives a better error message, by printing what performatives are being repeated. 

If you think it is not that useful I simply the code.